src/neuroconv/tools/nwb_helpers.py

```diff
@@ -47,6 +47,8 @@ def get_default_nwbfile_metadata() -> DeepDict:
 
 def make_nwbfile_from_metadata(metadata: dict) -> NWBFile:
     """Make NWBFile from available metadata."""
+    keys_to_keep = ["NWBFile", "Subject"]
+    metadata = {key: metadata[key] for key in keys_to_keep if key in metadata}
     metadata = dict_deep_update(get_default_nwbfile_metadata(), metadata)
     nwbfile_kwargs = metadata["NWBFile"]
```

Comment:
```suggestion
    nwbfile_kwargs = metadata["NWBFile"]
    if "session_description" not in nwbfile_kwargs:
        nwbfile_kwargs.update(session_description="No description.")
    if "identifier" not in nwbfile_kwargs:
        nwbfile_kwargs.update(identifier=uuid.uuid4())
```

Though, y'know, looking deeper at it all it's really doing is adding the UUID and description (only two fields...); maybe we could just do direct operations on it like so, which would make the function more readable for the future and avoids in-place modifications to the `metadata` object (not a fun experience for most people to magically modify arguments passed into functions)

Either that or assert that those fields are in the metadata before creating the file - your choice - but I think we can avoid deep updating altogether towards the goal of simplifying what should have already been a very simple function